### PR TITLE
HBASE-21002 make an assembly for hbase-connectors

### PR DESCRIPTION
**Pull request by hbasejanitor**
README for usage, assembly, some tweaks to get it working with Hbase 2.X.

### CODE REVIEW

File: kafka/hbase-kafka-proxy/src/main/java/org/apache/hadoop/hbase/kafka/KafkaProxy.java [hunk 1]
```diff
@@ -154,19 +150,28 @@ public static void main(String[] args) throws Exception {
...
         "enable peer on startup (defaults to false)");
 
-
     LOG.info("STARTING executorService " + HRegionServer.class.getSimpleName());
     VersionInfo.logVersion();
 
     Configuration conf = HBaseConfiguration.create();
     CommandLine commandLine = null;
+
+    Configuration commandLineConf = new Configuration();
```
[the remark this answers is not in the document]
Comment (hbasejanitor) [reply kafka/hbase-kafka-proxy/src/main/java/org/apache/hadoop/hbase/kafka/KafkaProxy.java]: Hi Stack, I'm doing it that way so the user can pass -Dconfkey=confValue on the command line.  What I do is create an empty conf object, then use GenericOptionsParser to pick out all the -D arguments.  I then feed those values into the region server that I start.

File: kafka/hbase-kafka-proxy/pom.xml [hunk 2]
```diff
@@ -76,43 +77,40 @@
...
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.yetus</groupId>
+      <artifactId>audience-annotations</artifactId>
+      <version>${audience-annotations.version}</version>
+    </dependency>
     <dependency>
       <groupId>org.apache.kafka</groupId>
       <artifactId>kafka-clients</artifactId>
```
[the remark this answers is not in the document]
Comment (hbasejanitor) [reply kafka/hbase-kafka-proxy/pom.xml]: I'll change this to hbase-shaded-client

File: kafka/README [hunk 3]
```diff
@@ -0,0 +1,126 @@
+Hbase Kafka Proxy
```
[the remark this answers is not in the document]
Comment (hbasejanitor) [reply kafka/README]: Yes, it should =-)

**Comment by hbasejanitor** [comment]
I think the exception was a classpath issue.  I've trimmed down the dependencies.

File: kafka/hbase-kafka-proxy/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (saintstack) [reply kafka/hbase-kafka-proxy/pom.xml]: This going to be done in a follow-up?

File: kafka/README (hunk 3 above)
[the remark this answers is not in the document]
Comment (saintstack) [reply kafka/README]: Duh

File: kafka/hbase-kafka-proxy/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (hbasejanitor) [reply kafka/hbase-kafka-proxy/pom.xml]: Sorry, I've been away from this patch for a while.

hbase-client isn't need at all, but hbase-server is.  Without it, I don't have access to the classes to start a region server.

[the remark this answers is not in the document]
Comment (saintstack) [reply kafka/hbase-kafka-proxy/pom.xml]: Ok. Makes sense.

**Comment by saintstack** [comment]
I applied the patch and was able to start up the proxy.

I did not test if it worked replicating out to kafka. I figure that can be done in follow-on post-commit. Stuff seems to basically work with basic doc. That is good enough for first commit.

**Pull request merged**